src/Editor/Editor.cpp

```diff
@@ -225,6 +227,94 @@ void Editor::Show(float deltaTime) {
     if (currentEntity != nullptr) {
         glm::mat4 currentEntityMatrix = currentEntity->GetLocalMatrix();
 
+
+        currentEntityMatrix = currentEntity->GetLocalMatrix();
+
+        // Projection matrix.
+        glm::mat4 projectionMatrix = cameraEntity->GetComponent<Component::Lens>()->GetProjection(glm::vec2(ImGui::GetWindowSize().x, ImGui::GetWindowSize().y));
+
+        // View matrix.
+        glm::mat4 viewMatrix = glm::inverse(cameraEntity->GetModelMatrix());
+
+        // Load Vertices and Indices for the ray-triangle intersection test when the "paint mode" button is pressed.
+        if (currentEntity->loadPaintModeClicked && currentEntity->vertsLoaded) {
+
+            Geometry::Model * model = dynamic_cast<Geometry::Model*>(currentEntity->GetComponent<Component::Mesh>()->geometry);
+            std::string modelPath = Hymn().GetPath() + FileSystem::DELIMITER + model->path + model->name + ".asset";
+            Geometry::AssetFileHandler handler;
+            handler.Open(modelPath.c_str());
+            handler.LoadMeshData(0);
+            Geometry::AssetFileHandler::MeshData* data = handler.GetStaticMeshData();
+
+            // Delete previously loaded verts and indices.
+            if (nrOfVertices > 0) {
+                delete[] vertices;
+                delete[] indices;
+                vertices = nullptr;
+                indices = nullptr;
+                nrOfVertices = 0;
+            }
+
+            vertices = new Video::Geometry::VertexType::StaticVertex[data->numVertices];
+            indices = new uint32_t[data->numIndices];
+
+            for (int i = 0; i < data->numVertices; i++)
+                vertices[i] = data->staticVertices[i];
+
+            for (int i = 0; i < data->numIndices; i++)
+                indices[i] = data->indices[i];
+
+            nrOfIndices = data->numIndices;
+            nrOfVertices = data->numVertices;
+
+            handler.Close();
+            currentEntity->vertsLoaded = false;
+        }
+
+        // Ray-Triangle intersection test.     
+        if (currentEntity->brushActive) {
+
+            glm::vec3 last_p0;
+            glm::vec3 last_p1;
+            glm::vec3 last_p2;
+            glm::vec3 p0;
+            glm::vec3 p1;
+            glm::vec3 p2;
+
+            mousePicker.Update();
+            mousePicker.UpdateProjectionMatrix(cameraEntity->GetComponent < Component::Lens>()->GetProjection(glm::vec2(MainWindow::GetInstance()->GetSize().x, MainWindow::GetInstance()->GetSize().y)));
```

Comment:
The spacing around the left angle bracket can be removed for improved readability.